columnflow/selection/cms/met_filters.py

```diff
@@ -0,0 +1,71 @@
+# coding: utf-8
+
+"""
+Selector related to MET filters.
+"""
+
+from __future__ import annotations
+
+from columnflow.selection import Selector, selector
+from columnflow.util import maybe_import
+from columnflow.columnar_util import Route
+
+
+ak = maybe_import("awkward")
+
+
+@selector(
+    uses={"event", "nFlag"},
+)
+def met_filters(
+    self: Selector,
+    events: ak.Array,
+    **kwargs,
+) -> ak.Array:
+    """
+    Compute a selection mask to filter out noisy/anomalous high-MET events (MET filters).
+
+    Individual filter decisions based on different criteria are stored as bool-valued columns
+    in the input NanoAOD. The columns to apply are specified via an auxiliary config entry:
+
+    .. code-block:: python
+
+        cfg.x.met_filters = {
+            "Flag.globalSuperTightHalo2016Filter",
+            "Flag.HBHENoiseFilter",
+            "Flag.HBHENoiseIsoFilter",
+            "Flag.EcalDeadCellTriggerPrimitiveFilter",
+            "Flag.BadPFMuonFilter",
+            "Flag.BadPFMuonDzFilter",
+            "Flag.eeBadScFilter",
+            "Flag.ecalBadCalibFilter",
+        }
+
+    The specified columns are interpreted as booleans, with missing values treated as *True*,
+    i.e. the event is considered to have passed the corresponding filter.
+
+    Returns a bool array containing the logical AND of all input columns.
+    """
+    met_filters = set(self.config_inst.x.met_filters)
+
+    result = ak.ones_like(events.event, dtype=bool)
+
+    for route in met_filters:
+        # interpret column values as booleans
+        vals = Route(route).apply(events)
+        vals = ak.values_astype(vals, bool)
+        # treat missing values as a "pass"
+        vals = ak.fill_none(vals, True)
+        # append to the result
+        result = (result & vals)
+
+    return result
+
+
+@met_filters.init
+def met_filters_init(self: Selector) -> None:
+    """
+    Read MET filters from config and add them as input columns.
+    """
+    met_filters = set(self.config_inst.x.met_filters)
```

Comment:
One thing here: In a previous analysis we had different filters being applied to either data or MC. I think the `cfg.x.met_filters` should be able to be a dictionary with keys `data` and `mc`, and if that is the case, the logic at this point should pick the correct ones based on `self.dataset_inst.data_source`. Suggestion:

```python
met_filters = self.config_inst.x.met_filters
if isinstance(met_filters, dict):
    # do nothing when no dataset_inst is known
    if not getattr(self, "dataset_inst", None):
        return
    met_filters = met_filters[self.dataset_inst.data_source]

# store filters as an attribute for faster lookup
self.met_filters = set(met_filters)

self.uses |= self.met_filters
```

Reply:
Very good point, there needs to be a distinction here. I quite like the suggestion as well since it handles some edge cases quite nicely!